src/store/slices/userSlice/userSlice.ts

```diff
@@ -225,97 +344,68 @@ const mainReducer = createSlice({
       state.users = [];
       state.isLoading = false;
       state.errorMessage = '';
     });
-    builder.addCase(logOutUserThank.rejected, (state) => {
-      state.isLoading = false;
-    });
-    builder.addCase(logInUserThank.pending, (state) => {
-      state.isLoading = true;
-      state.errorMessage = '';
-    });
-    builder.addCase(logInUserThank.fulfilled, (state) => {
-      state.isLoading = false;
-      state.errorMessage = '';
-    });
-    builder.addCase(logInUserThank.rejected, (state) => {
-      state.isLoading = false;
-    });
-    builder.addCase(updateUserThank.pending, (state) => {
-      state.isLoading = true;
-      state.errorMessage = '';
-    });
-    builder.addCase(updateUserThank.fulfilled, (state) => {
-      state.isLoading = false;
-      state.errorMessage = '';
-    });
-    builder.addCase(updateUserThank.rejected, (state) => {
-      state.isLoading = false;
-    });
-    builder.addCase(getLimitUsersThank.pending, (state) => {
-      state.errorMessage = '';
-    });
-
     builder.addCase(getLimitUsersThank.fulfilled, (state, action) => {
       state.users = action.payload.usersData;
       state.lastUser = action.payload.lastVisible;
       state.lengthDataUsers = action.payload.lengthData!;
     });
-    builder.addCase(getLimitUsersThank.rejected, (state) => {
-      state.isLoading = false;
-    });
-    builder.addCase(getMoreUsersThank.pending, (state) => {
-      state.errorMessage = '';
-    });
-
     builder.addCase(getMoreUsersThank.fulfilled, (state, action) => {
       state.users = [...state.users, ...action.payload.usersData];
       state.lastUser = action.payload.lastVisible;
     });
-    builder.addCase(getMoreUsersThank.rejected, (state) => {
+    builder.addCase(isOccupiedNickThank.fulfilled, (state) => {
+      state.errorMessage = '';
+    });
+    builder.addCase(changeUserPhotoThank.fulfilled, (state, action) => {
+      state.isLoading = false;
+      state.photoURL = action.payload!;
+    });
+    builder.addCase(changeUserPhotoThank.rejected, (state) => {
       state.isLoading = false;
+      state.errorMessage = ErrorUpload.Size;
     });
-    builder.addCase(setUserThank.pending, (state) => {
+    builder.addMatcher(isPendingAction, (state) => {
       state.isLoading = true;
       state.errorMessage = '';
     });
-    builder.addCase(setUserThank.fulfilled, (state) => {
-      state.isLoading = false;
+    builder.addMatcher(isPendingActionWithoutLoader, (state) => {
       state.errorMessage = '';
     });
-    builder.addCase(setUserThank.rejected, (state) => {
+    builder.addMatcher(isFulfilledAction, (state) => {
       state.isLoading = false;
-    });
-    builder.addCase(isOccupiedNickThank.pending, (state) => {
       state.errorMessage = '';
     });
-    builder.addCase(isOccupiedNickThank.fulfilled, (state) => {
-      state.errorMessage = '';
+    builder.addMatcher(isRejectedAction, (state) => {
+      state.isLoading = false;
```

Comment:
how we handle the error message?
which comes from:
`return rejectWithValue(message);`